create-presigned-download-url/src/main/java/no/unit/nva/download/publication/file/publication/RestPublicationService.java

```diff
@@ -58,44 +57,70 @@ public RestPublicationService(HttpClient client, ObjectMapper objectMapper, Stri
     @JacocoGenerated
     public RestPublicationService(Environment environment) {
         this(HttpClient.newHttpClient(), JsonUtils.objectMapper, environment.readEnv(API_SCHEME_ENV),
-                environment.readEnv(API_HOST_ENV));
+            environment.readEnv(API_HOST_ENV));
     }
 
     /**
      * Retrieve publication metadata.
      *
-     * @param identifier    identifier
-     * @param authorization authorization
+     * @param identifier         identifier
+     * @param authorizationToken authorization
      * @return A publication
      * @throws ApiGatewayException exception thrown if value is missing
      */
-    public Publication getPublication(UUID identifier, String authorization) throws ApiGatewayException {
-        URI uri = UrlBuilder.empty()
-                .withScheme(apiScheme)
-                .withHost(apiHost)
-                .withPath(PATH + identifier.toString())
-                .toUri();
-
-        HttpRequest httpRequest = HttpRequest.newBuilder()
-                .uri(uri)
-                .header(HttpHeaders.ACCEPT, APPLICATION_JSON)
-                .header(HttpHeaders.AUTHORIZATION, authorization)
-                .GET()
-                .build();
+    public Publication getPublication(UUID identifier, String authorizationToken)
+        throws ApiGatewayException {
 
-        try {
-            HttpResponse<String> httpResponse = client.send(httpRequest, HttpResponse.BodyHandlers.ofString());
+        URI uri = buildUriToPublicationService(identifier);
+        HttpRequest httpRequest = buildHttpRequest(uri, authorizationToken);
+        return fetchPublicationFromService(identifier, uri, httpRequest);
+    }
 
-            if (httpResponse.statusCode() == SC_NOT_FOUND) {
-                throw new NotFoundException(ERROR_PUBLICATION_NOT_FOUND_FOR_IDENTIFIER + identifier);
-            }
 
-            JsonNode jsonNode = objectMapper.readTree(httpResponse.body());
-            return objectMapper.convertValue(jsonNode, Publication.class);
 
+    private Publication fetchPublicationFromService(UUID identifier, URI uri, HttpRequest httpRequest)
+        throws NoResponseException {
+        try {
+            return fetchPublicationFromService(identifier, httpRequest);
         } catch (Exception e) {
             throw new NoResponseException(ERROR_COMMUNICATING_WITH_REMOTE_SERVICE + uri.toString(), e);
         }
     }
 
+    private Publication fetchPublicationFromService(UUID identifier, HttpRequest httpRequest)
+        throws java.io.IOException, InterruptedException, NotFoundException {
+
+        HttpResponse<String> httpResponse = sendHttpRequest(httpRequest);
+        if (httpResponse.statusCode() == SC_NOT_FOUND) {
+            throw new NotFoundException(ERROR_PUBLICATION_NOT_FOUND_FOR_IDENTIFIER + identifier);
```

Comment:
Could we simply bubble the `problem response` ? And maybe decorate that it passed via this lambda?

Reply:
We cannot do directly this because the exception will generate a new Problem. So we will have a Problem serialized as a details in a Problem. I have committed some changes where I propagate the error message from the service. However, I do am not sure about it, as the only thing we will be saying is that the Publication cannot be found.